lms/djangoapps/support/static/support/jsx/entitlements/components/Search/Search.jsx

```diff
@@ -0,0 +1,36 @@
+import React from 'react';
+import { InputText } from '@edx/paragon';
+
+
+class Search extends React.Component{
+  constructor(props) {
+    super(props)
+    this.state = { username: '' }
+  }
+
+  handleSubmit(event) {
+    event.preventDefault();
+    //since the below state update will cause react to re-render dom, the default refresh is unneeded
+    this.props.fetchEntitlements(this.state.username);
+  }
+
+  handleUserChange(event) {
+    this.setState({ username: event });
+  }
+
+  render() {
+    return (
+      <form onSubmit={ this.handleSubmit.bind(this) }>
```

Comment:
Does this work or do you need to assign an instance variable in the constructor that references the bound method?

```javascript
constructor(props) {
  super(props);
  
  this.handleSubmit = this.handleSubmit.bind(this);
}
```

Reply:
This works, the downside is I'd have to bind(this) everywhere that I use handleSubmit. But given that I only use it once I sort of feel like this is as clean as defining it in the constructor, thoughts?

Reply:
`.bind()` creates a new function.

From [the Handling Events part of the React Documentation](https://reactjs.org/docs/handling-events.html)

> The problem with this syntax is that a different callback is created each time the LoggingButton renders. In most cases, this is fine. However, if this callback is passed as a prop to lower components, those components might do an extra re-rendering. We generally recommend binding in the constructor or using the class fields syntax, to avoid this sort of performance problem.

I agree that this shouldn't be an issue most of the time and that it adds some bloat, but if there's generally agreed upon safety when `bind`ing in the `constructor` and it doesn't seem to create *that* much overhead I think there isn't much downside to `bind`ing in the `constructor`.